src/scripting/lua_styles.cc

```diff
@@ -0,0 +1,233 @@
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: open_p()
+
+   Return richtext formatting to start a paragraph with the given style.
+
+   :type style: class:`string`
+   :arg style: name of the paragraph style to use.
+
+   :returns: The richtext formatting that starts a paragraph.
+*/
+static int L_open_p(lua_State* L) {
+	const std::string style(luaL_checkstring(L, 1));
+	const std::string result = g_style_manager->paragraph_style(style).open_paragraph();
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: open_p_with_attr(attributes)
+
+   Return richtext formatting to start a paragraph with the given style and extra attributes.
+
+   :type style: class:`string`
+   :arg style: name of the paragraph style to use.
+   :type attributes: class:`string`
+   :arg attributes: attributes for the paragraph tag.
+
+   :returns: The richtext formatting that starts the paragraph.
+*/
+static int L_open_p_with_attr(lua_State* L) {
+	const std::string style(luaL_checkstring(L, 1));
+	const std::string attr(luaL_checkstring(L, 2));
+	const std::string result = g_style_manager->paragraph_style(style).open_paragraph(attr);
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: close_p()
+
+   Return richtext formatting to close a paragraph of the given style.
+
+   :type style: class:`string`
+   :arg style: name of the paragraph style to use.
+
+   :returns: The richtext formatting that closes a paragraph.
+*/
+static int L_close_p(lua_State* L) {
+	const std::string style(luaL_checkstring(L, 1));
+	const std::string result = g_style_manager->paragraph_style(style).close_paragraph();
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: as_font_from_p(p_style, text)
+
+   Format the given text with the font style of the given paragraph style, without the paragraph
+   tags.
+
+   :type style: class:`string`
+   :arg p_style: name of the paragraph style to use.
+   :type text: class:`string`
+   :arg text: text to format.
+
+   :returns: The text with richtext tags.
+*/
+static int L_as_font_from_p(lua_State* L) {
+	const std::string p_style(luaL_checkstring(L, 1));
+	const std::string text(luaL_checkstring(L, 2));
+	const std::string result = g_style_manager->paragraph_style(p_style).font().as_font_tag(text);
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: color(style_name)
+
+   Return the given color style in hex format suitable for font tags.
+
+   :type style_name: class:`string`
+   :arg style_name: name of the color style to use.
+
+   :returns: The color as 6 characters hexadecimal RGB value.
+*/
+static int L_color(lua_State* L) {
+	const std::string style_name(luaL_checkstring(L, 1));
+	const std::string result = g_style_manager->color(style_name).hex_value();
+	lua_pushstring(L, result.c_str());
+	return 1;
+}
+
+/* RST
+.. function:: dimension(style_name)
```

Comment:
Maybe consider a different name? To me "dimension" sounds like "extent", i.e. 2-dimensional width and height. Maybe it's just that I work too much with Java where Dimension is a much-used class that represents an Extent, but IMHO "gap" or "spacing" or "padding" or "size" would sound less confusing.

Reply:
Originally this was `size`, but I was afraid it may clash or could be confused with standard functions. Can't name it `gap`, `spacing` or `padding` because it's already used for window sizes as well, and may be used for other UI element sizes when we convert all C++ code.

OK, then how about `styled_size` which is already the name of the enum?